velox/dwio/parquet/tests/ParquetTpchTestBase.h

```diff
@@ -48,18 +53,44 @@ class ParquetTpchTest : public testing::Test {
     functions::prestosql::registerAllScalarFunctions();
     parse::registerTypeResolver();
     filesystems::registerLocalFileSystem();
-    registerParquetReaderFactory();
+    registerParquetReaderFactory(parquetReaderType_);
 
     auto hiveConnector =
         connector::getConnectorFactory(
             connector::hive::HiveConnectorFactory::kHiveConnectorName)
             ->newConnector(kHiveConnectorId, nullptr);
     connector::registerConnector(hiveConnector);
-    tempDirectory_ = exec::test::TempDirectoryPath::create();
+    tempDirectory_ = TempDirectoryPath::create();
     saveTpchTablesAsParquet();
     tpchBuilder_.initialize(tempDirectory_->path);
+    isReaderFactoryRegistered_ = true;
+  }
+
+  void SetUp() {
```

Comment:
With `SetUpTestSuite()` I don't think `SetUp` is required.

Reply:
SetUp is retained to register the proper Parquet reader for each reader in a TEST_P, since SetUpTestSuite is called only once for the instantiation.

Reply:
We should be able to call the parquet reader registration only in `SetUp` then?
The if-else must be avoided.

Reply:
Thanks for the suggestion, I tried that but tpchBuilder_.initialize() requires the reader registration to be done and since SetUp is called after SetUpTestSuite, it is necessary to register the reader the very first time when SetUpTestSuite is called and change the registration for reader type in every call to SetUp as well.

Reply:
This will not work when the tests run concurrently correct?
So these objects should not be static.

Reply:
Yes, when the tests run concurrently, this will not work. When I ran the suite locally, the tests ran sequentially so I did not account for that, I will revert the static changes and move the initialization back to the SetUp function, thanks!